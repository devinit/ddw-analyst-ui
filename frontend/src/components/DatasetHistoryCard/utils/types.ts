export interface SavedQueryData {
  id: number;
  saved_query_db_table: string;
  operation: number;
  full_query: string | null;
  active: boolean;
  description: string;
  created_on: string;
  user: string;
  status: string;
}
