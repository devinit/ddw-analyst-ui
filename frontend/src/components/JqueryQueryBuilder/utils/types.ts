export type JqueryQueryBuilderFilter = {
  condition?: string;
  rules?: (JqueryQueryBuilderFilterComparator | JqueryQueryBuilderFilter)[];
};

export type JqueryQueryBuilderHaving = {
  condition?: string;
  rules?: (JqueryQueryBuilderHavingComparator | JqueryQueryBuilderHaving)[];
};

export type JqueryQueryBuilderHavingComparator = {
  id: string;
  field: string;
  type: string;
  input: string;
  operator: JqueryQueryBuilderComps;
  value?: string | number | { plain: string | number } | { column: string; aggregate: string };
  values?: { value: string; label: string }[];
};

export type JqueryQueryBuilderFilterComparator = {
  id: string;
  field: string;
  type: string;
  input: string;
  operator: JqueryQueryBuilderComps;
  value: string | number;
};
export type JqueryQueryBuilderComps =
  | 'equal'
  | 'not_equal'
  | 'less'
  | 'less_or_equal'
  | 'greater'
  | 'greater_or_equal'
  | 'between'
  | 'in';

export type JqueryQueryBuilderFieldData = {
  id: string | number;
  label: string;
  type: string;
};

export type JqueryQueryBuilderIcons = {
  add_group: string;
  add_rule: string;
  remove_group: string;
  remove_rule: string;
  error: string;
};
