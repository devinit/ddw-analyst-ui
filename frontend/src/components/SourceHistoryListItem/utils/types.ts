export interface FrozenData {
  id: number;
  parent_db_table: string;
  frozen_db_table: string | null;
  active: boolean;
  description: string;
  created_on: string;
  user: string;
  status: string;
  logs: string | null;
}
