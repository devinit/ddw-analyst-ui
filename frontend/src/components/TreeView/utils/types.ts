import { ReactElement } from 'react';

export interface TreeViewProps {
  data: Data[];
  depth?: number;

  deleteElement?: ReactElement;

  getStyleClassCb?: (node: Data) => void;

  isCheckable?: (node: Data, depth?: number) => void;
  isDeletable?: (node: Data, depth?: number) => void;
  isExpandable?: (node: Data, depth?: number) => void;

  keywordChildren?: string;
  keywordChildrenLoading?: string;
  keywordKey?: string;
  keywordLabel?: string;

  loadingElement?: ReactElement;
  noChildrenAvailableMessage?: string;

  onCheckToggleCb?: (data: Data[], depth?: number) => void;
  onDeleteCb?: (node: Data, data: Data[], depth?: number) => void;
  onExpandToggleCb?: (node: Data, depth?: number) => void;
  onUpdateCb?: (data: unknown[], depth?: number) => void;

  transitionEnterTimeout?: number;
  transitionExitTimeout?: number;
}

export interface Data {
  id: number;
  name: string;
  isExpanded?: boolean;
  isChecked?: boolean;
  isChildrenLoading?: boolean;
  children?: Data[];
}
