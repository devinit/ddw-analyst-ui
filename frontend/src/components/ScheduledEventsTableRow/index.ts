export * from './ScheduledEventsTableRow';
export * from './utils/dateConversion';
