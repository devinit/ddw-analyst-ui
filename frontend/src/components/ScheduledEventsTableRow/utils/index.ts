import axios from 'axios';
import * as localForage from 'localforage';
import { Moment } from 'moment';
import { api, localForageKeys } from '../../../utils';

interface RunInstancePayload {
  start_at: Moment;
  status: string;
}

interface RunInstance {
  scheduled_event: number;
  start_at: string;
  ended_at: string | undefined | null;
  status: string;
}

interface ServerResponse {
  error: string;
  success: string;
  result: RunInstance[];
}

interface RunInstanceResponse {
  status: number;
  statusText: string;
  data: ServerResponse;
}

export const createRunInstance = async (
  scheduleId: number,
  data: RunInstancePayload,
): Promise<RunInstanceResponse> => {
  const token = await localForage.getItem<string>(localForageKeys.API_KEY);

  return await axios.request({
    url: api.routes.CREATE_SCHEDULED_INSTANCE.replace('{scheduleId}', `${scheduleId}`),
    method: 'post',
    headers: {
      'Content-Type': 'application/json',
      Authorization: `token ${token}`,
    },
    data,
  });
};

const intervalType: { [char: string]: string } = {
  min: 'Minutes',
  sec: 'Seconds',
  hrs: 'Hours',
  dys: 'Days',
  wks: 'Weeks',
  mnt: 'Months',
  yrs: 'Years',
};

export const convertIntervalType = (value: string): string | undefined => {
  for (const key in intervalType) {
    if (key === value) {
      return intervalType[key];
    }
  }
};
