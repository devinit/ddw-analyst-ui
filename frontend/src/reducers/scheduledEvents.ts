import { List, Map, fromJS } from 'immutable';
import { Action, Reducer } from 'redux';
import { ScheduledEvent, ScheduledEventMap } from '../types/scheduledEvents';

export interface ScheduledEventsAction extends Action {
  scheduledEvents?: ScheduledEvent[];
  activeScheduledEvent?: ScheduledEventMap;
  count: number;
  activeScheduleEventIndex?: number;
  payload: Partial<{
    limit: number;
    offset: number;
    link: string;
    search: string;
    id: number | string;
  }>;
  loading?: boolean;
}
interface State {
  loading: boolean;
  scheduledEvents: List<ScheduledEventMap>;
  activeScheduledEvent?: ScheduledEventMap;
  count: number;
  activeOperationId: number;
}
export type ScheduledEventsState = Map<keyof State, State[keyof State]>;

const prefix = 'scheduledEvents';
export const FETCH_SCHEDULED_EVENTS = `${prefix}.FETCH`;
export const FETCH_SCHEDULED_EVENTS_SUCCESSFUL = `${prefix}.FETCH_SUCCESSFUL`;
export const FETCH_SCHEDULED_EVENTS_FAILED = `${prefix}.FETCH_FAILED`;
export const SET_ACTIVE_SCHEDULED_EVENT = `${prefix}.SET_ACTIVE_SCHEDULED_EVENT`;
export const FETCH_SCHEDULED_EVENT = `${prefix}.FETCH_SCHEDULED_EVENT`;
export const FETCH_SCHEDULED_EVENT_FAILED = `${prefix}.FETCH_FAILED`;
export const DELETE_SCHEDULED_EVENT = `${prefix}.DELETE_SCHEDULED_EVENT`;
export const DELETE_SCHEDULED_EVENT_SUCCESSFUL = `${prefix}.DELETE_SCHEDULED_EVENT_SUCCESSFUL`;
export const DELETE_SCHEDULED_EVENT_FAILED = `${prefix}.DELETE_SCHEDULED_EVENT_FAILED`;

export const defaultState: ScheduledEventsState = fromJS({
  loading: false,
  scheduledEvents: [],
  activeScheduledEventId: 1,
  limit: 10,
  offset: 0,
});

export const scheduledEventsReducer: Reducer<ScheduledEventsState, ScheduledEventsAction> = (
  state = defaultState,
  action,
) => {
  if (action.type === FETCH_SCHEDULED_EVENTS || action.type === FETCH_SCHEDULED_EVENT) {
    return state.set('loading', true);
  }
  if (action.type === FETCH_SCHEDULED_EVENTS_SUCCESSFUL && action.scheduledEvents) {
    return state.withMutations((map) =>
      map
        .set('loading', false)
        .set('scheduledEvents', fromJS(action.scheduledEvents))
        .set('count', action.count),
    );
  }
  if (
    action.type === FETCH_SCHEDULED_EVENTS_FAILED ||
    action.type === FETCH_SCHEDULED_EVENT_FAILED
  ) {
    return state.set('loading', false);
  }
  if (action.type === SET_ACTIVE_SCHEDULED_EVENT) {
    return state.set('activeScheduledEvent', action.activeScheduledEvent).set('loading', false);
  }

  return state;
};
