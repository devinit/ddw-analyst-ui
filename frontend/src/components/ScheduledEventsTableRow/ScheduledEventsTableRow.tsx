import moment from 'moment';
import React, { FunctionComponent, useContext, useState } from 'react';
import { Button } from 'react-bootstrap';
import { ScheduledEventContext } from '../../pages/ScheduledEvents';
import { ScheduledEvent } from '../../types/scheduledEvents';
import { convertIntervalType, createRunInstance } from './utils';

export interface ScheduledEventsTableRowProps {
  id: number;
  event: ScheduledEvent;
  classNames?: string;
}

export const ScheduledEventsTableRow: FunctionComponent<ScheduledEventsTableRowProps> = (props) => {
  const { setActiveEvent } = useContext(ScheduledEventContext);
  const [isCreatingInstance, setIsCreatingInstance] = useState(false);

  const onRunNow = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>): void => {
    event.stopPropagation();
    setIsCreatingInstance(true);
    createRunInstance(props.event.id, {
      start_at: moment(), // eslint-disable-line @typescript-eslint/camelcase
      status: 'p',
    })
      .then((response) => {
        if (response.status === 201 || response.status === 200) {
          setIsCreatingInstance(false);
        } else {
          console.log(JSON.stringify(response));
        }
      })
      .catch((error) => {
        console.log(JSON.stringify(error));
        setIsCreatingInstance(false);
      });
  };
  const onRowClick = (): void => {
    if (setActiveEvent) setActiveEvent(props.event);
  };

  return (
    <tr onClick={onRowClick} className={props.classNames}>
      <td className="text-center">{props.id}</td>
      <td>{props.event.name}</td>
      <td>{props.event.description}</td>
      <td>
        <div className="togglebutton">
          <label className="enabled">
            <input type="checkbox" defaultChecked={props.event.enabled} disabled />
            <span className="toggle"></span>
          </label>
        </div>
      </td>
      <td>
        {moment(props.event.start_date).format('LL')} at{' '}
        {moment.utc(props.event.start_date).format('LT')}
      </td>
      <td>
        {props.event.repeat && props.event.interval && props.event.interval_type
          ? `Every ${props.event.interval} ${convertIntervalType(props.event.interval_type)}`
          : 'None'}
      </td>
      <td>
        <Button variant="outline-danger" size="sm" onClick={onRunNow} disabled={isCreatingInstance}>
          {isCreatingInstance ? 'Creating instance...' : 'Run Now'}
        </Button>
      </td>
    </tr>
  );
};
