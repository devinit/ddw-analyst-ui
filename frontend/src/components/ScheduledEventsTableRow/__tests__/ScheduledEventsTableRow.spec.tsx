import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { ScheduledEventsTableRow } from '../ScheduledEventsTableRow';
import { cleanup } from '@testing-library/react';
import { ScheduledEvent } from '../../../types/scheduledEvents';

const currentDate = new Date();

const event: ScheduledEvent = {
  id: 1,
  name: 'Common Reporting Standard',
  description: 'event1',
  enabled: true,
  interval: 1,
  interval_type: 'sec',
  repeat: true,
  start_date: currentDate.toISOString(),
};

afterEach(cleanup);

test('renders correctly with default props', () => {
  const renderer = TestRenderer.create(<ScheduledEventsTableRow id={1} event={event} />).toJSON();

  expect(renderer).toMatchSnapshot();
});
