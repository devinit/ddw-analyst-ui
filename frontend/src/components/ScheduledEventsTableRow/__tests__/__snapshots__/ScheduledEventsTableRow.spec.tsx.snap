// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with default props 1`] = `
<tr
  onClick={[Function]}
>
  <td
    className="text-center"
  >
    1
  </td>
  <td>
    Common Reporting Standard
  </td>
  <td>
    event1
  </td>
  <td>
    <div
      className="togglebutton"
    >
      <label
        className="enabled"
      >
        <input
          defaultChecked={true}
          disabled={true}
          type="checkbox"
        />
        <span
          className="toggle"
        />
      </label>
    </div>
  </td>
  <td>
    August 12, 2020
     at
     
    8:30 AM
  </td>
  <td>
    Every 1 Seconds
  </td>
  <td>
    <button
      className="btn btn-outline-danger btn-sm"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Run Now
    </button>
  </td>
</tr>
`;
